windows/hid.c

```diff
@@ -248,6 +285,23 @@ static int lookup_functions()
 	else
 		return -1;
 
+	cfgmgr32_lib_handle = LoadLibraryA("cfgmgr32.dll");
+	if (cfgmgr32_lib_handle) {
+#if defined(__GNUC__)
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wcast-function-type"
+#endif
+#define RESOLVE(x) x = (x##_)GetProcAddress(cfgmgr32_lib_handle, #x);
+		RESOLVE(CM_Locate_DevNodeW);
+		RESOLVE(CM_Get_Parent);
+		RESOLVE(CM_Get_DevNode_PropertyW);
+		RESOLVE(CM_Get_Device_Interface_PropertyW);
+#undef RESOLVE
+#if defined(__GNUC__)
+# pragma GCC diagnostic pop
+#endif
+	}
```

Comment:
I initialized those function pointers as `NULL` instead.

Reply:
_strictly_ speaking, that's not enough:

There is a possible/impossible scenario where it can fail:
1) hid_init - cfgmgr32 is OK;
2) hid_deinit
3) Maybe the runtime env has changed
4) hid_init - cfgmgr32 didn't load, but `CM_...` pointers are dangling from last hid_init

Reply:
What I suggest is:
```c
if (cfgmgr32_lib_handle) {
    RESOLVE(CM_Locate_DevNodeW);
    // ...
}
else {
    CM_Locate_DevNodeW = NULL;
    // ...
}
```

Reply:
done